### DISCUSSION TITLE
Windows. The filename or extension is too long

### DISCUSSION THREAD

**Discussion by akhvatov21** [ascopes/protobuf-maven-plugin#480] [Q&A]
Hi,

Windows.

I'm trying to use this plugin but I am having this problem

`Caused by: java.io.IOException: Cannot run program "C:\Users\akhvatov\.m2\repository\com\google\protobuf\protoc\4.29.0\protoc-4.29.0-windows-x86_64.exe": CreateProcess error=206, The filename or extension is too long
    at java.lang.ProcessBuilder.start (ProcessBuilder.java:1170)
    at java.lang.ProcessBuilder.start (ProcessBuilder.java:1089)
    at io.github.ascopes.protobufmavenplugin.protoc.CommandLineExecutor.run (CommandLineExecutor.java:71)`

Is there a way to `shorten` the command line like in other plugins?

Thank you

**Comment by ascopes** [comment]
Which Windows version and Java version are you using? Asking since CI runs against Windows, so it'd be good to understand why this is failing.

**Reply by akhvatov21** [reply]
Windows 10 and Java 21

**Reply by akhvatov21** [reply]
For example in `xolstice` plugin there is a parameter `useArgumentFile`

**Reply by ascopes** [reply]
Thanks...

If you edit the registry key `HKLM\SYSTEM\CurrentControlSet\ControlFileSystem` for `LongPathsEnabled` (DWORD) to be the value `1`, and then reboot... does the issue still persist?

Just want to understand if it is Windows rejecting it on Java rejecting it on behalf of Windows, or whether it is the length of the commandline arguments that it is complaining about.

**Reply by akhvatov21** [reply]
I think it's a known issue with Windows, it allows the command line to be a maximum of  8191. So the workaround is using a manifest file, for example, IntelliJIdea is doing that if you have too big classpath for a test run

> If you edit the registry key HKLM\SYSTEM\CurrentControlSet\ControlFileSystem for LongPathsEnabled (DWORD) to be the value 1, and then reboot... does the issue still persist?

I'm not sure I would be able to do that, because I have my company laptop, probably it's forbidden

**Reply by ascopes** [reply]
Gotcha, ok... can you raise this as a bug on the issues page and include the requested logs it asks for on that page? I'll see if I can get to the bottom of it when I have time

**Comment by ascopes** [comment]
Closing this so the discussion moves to the GH-481 issue raised for this! I'll put any further updates over there.

Thanks
